### PR TITLE
[linux] Fix text selection via Shift+Home/End

### PR DESCRIPTION
**Pull request by jpnurmi**
This PR fixes text selection when pressing Shift + Home/End in a Flutter text field.

Fixes: flutter/flutter#76785

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.
- [ ] The reviewer has submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

**Comment by jpnurmi** [comment]
| Before | After | Native |
|---|---|---|
|  before | after | gedit |

**Review by chinmaygarde — APPROVED** [review]
cc @cbracken

File: shell/platform/common/text_input_model.h [hunk 1]
```diff
@@ -177,6 +193,14 @@ class TextInputModel {
   bool composing() const { return composing_; }
 
  private:
+  enum MoveOrSelect { Move, Select };
```
Comment (cbracken) [inline shell/platform/common/text_input_model.h:196]: ~~Two quick requests:~~
~~1. `enum class MoveOrSelect {`~~
~~2. Prefix the enum values with `k` so `kMove` and `kSelect`.~~

Actually, I think the code will be clearer if we get rid of this enum and the MoveOrSelect* methods and simply implement these directly inthe Move* and Select* methods. That avoids an extra level of indirection and the condition check.

**Review by cbracken — CHANGES_REQUESTED** [review]
Thanks for your PR -- one quick request, otherwise lgtm!

File: shell/platform/common/text_input_model.h (hunk 1 above)
Comment (jpnurmi) [reply shell/platform/common/text_input_model.h]: Thanks, done

**Review by cbracken — APPROVED** [review]
Fantastic! Thanks for the fix @jpnurmi!

LGTM stamp from a Japanese personal seal

**Pull request merged**